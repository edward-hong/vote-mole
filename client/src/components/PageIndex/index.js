import React from 'react'
import Pagination from 'react-ultimate-pagination-bootstrap-4'
import { isNil, and, divide } from 'ramda'

const PageIndex = ({ polls, count, pageSize, page, onPageChange }) => {
	if (and(isNil(polls), isNil(count))) {
		return null
	}

	return (
		<Pagination
			currentPage={page}
			totalPages={Math.ceil(divide(count, pageSize))}
			onChange={onPageChange}
		/>
	)
}

export default PageIndex
