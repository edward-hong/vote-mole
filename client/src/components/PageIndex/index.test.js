import React from 'react'
import { render } from '@testing-library/react'

import PageIndex from '.'

describe('PageIndex', () => {
	it('returns null if `poll` and `count` prop is undefined', () => {
		const { container } = render(
			<PageIndex
				polls={undefined}
				count={undefined}
				page={0}
				pageSize={5}
				onPageChange={() => {}}
			/>,
		)

		expect(container.firstChild).toBeNull()
	})

	it('returns null if `count` prop is 0', () => {
		const { container } = render(
			<PageIndex
				polls={[]}
				count={0}
				page={0}
				pageSize={5}
				onPageChange={() => {}}
			/>,
		)

		expect(container.firstChild).toBeNull()
	})

	it('renders correctly when `polls` prop is passed', () => {
		const polls = [
			{
				id: '5cdaf28cee882077685935e3',
				pollQuestion: 'PHP, Python or Ruby?',
			},
			{
				id: '5cdaf0d4ee882077685935de',
				pollQuestion: 'Favourite JavaScript Framework',
			},
			{
				id: '5cdaf0a6ee882077685935d9',
				pollQuestion: 'Favourite Sport',
			},
			{
				id: '5cdaf088ee882077685935d6',
				pollQuestion: 'Favourite Movie',
			},
			{
				id: '5cdaf04fee882077685935d1',
				pollQuestion: 'Favourite Superhero',
			},
		]
		const { container } = render(
			<PageIndex
				polls={polls}
				count={polls.length}
				page={0}
				pageSize={5}
				onPageChange={() => {}}
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})
})
