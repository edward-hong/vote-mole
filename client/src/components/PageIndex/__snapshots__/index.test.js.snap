// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageIndex renders correctly when \`polls\` prop is passed 1`] = `
.c0 .pagination {
  margin-top: 2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .page-link {
  color: #21252a;
  border-color: #21252a;
}

.c0 .page-link:focus,
.c0 .page-link:hover {
  color: white;
  border-color: #21252a;
  background-color: rgba(33,37,41,0.7);
  -webkit-transition: background-color 0.15s ease-in-out,color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out,color 0.15s ease-in-out;
}

.c0 .page-item.active .page-link {
  background-color: #21252a;
  border-color: #21252a;
}

<div
  class="c0"
>
  <ul
    class="pagination"
  >
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
      >
        «
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
      >
        ‹
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
      >
        ›
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
      >
        »
      </a>
    </li>
  </ul>
</div>
`;
